src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java

```diff
@@ -49,41 +54,58 @@ public MVNPEntityListener(MultiverseNetherPortals plugin) {
         this.worldManager = this.plugin.getCore().getMVWorldManager();
         this.pt = new PermissionTools(this.plugin.getCore());
         this.playerErrors = new HashMap<String, Date>();
-        this.eventRecord = new HashMap<String, Location>();
         this.messaging = this.plugin.getCore().getMessaging();
         this.locationManipulation = this.plugin.getCore().getLocationManipulation();
-
+        this.eventRecord = new HashMap<>();
+        this.eventRecord.put(PortalType.ENDER, new HashSet<>());
+        this.eventRecord.put(PortalType.NETHER, new HashSet<>());
     }
 
-    protected void shootPlayer(Player p, Block block, PortalType type) {
+    protected boolean shootPlayer(Player p, Block block, PortalType type) {
```

Comment:
Would you mind adding a javadoc here to help understand what the return value means?

Reply:
Sure, and I'll try to refactor it while I'm at it.

Reply:
I couldn't really find a good way to refactor it :/

But it's got a Javadoc now. Also, I lowered the strength because it was a bit too strong imo.